wallet-new/docs/updates.md

```diff
@@ -0,0 +1,656 @@
+    in using Daedalus.
+
+On the other hand, `wsuUpdateState` records the users' decision towards the update:
+
+    data UpdateState = Applied
+                     -- ^ The update has been fully applied.
+                     | Downloaded !Sha256Checksum !FilePath
+                     -- ^ The update has been downloaded but not applied yet. A Checksum
+                     -- can be stored to verify the integrity of the download.
+                     | Skipped
+                     -- ^ The update has been skipped by the user.
+                     | Proposed
+                     -- ^ The update has been just proposed and is waiting for users' feedback.
+
+Note how we could add an intermediate `Downloading` state, but this would complicate state-handling
+in case a user hard-quits Daedalus whilst the upgrade is being downloaded. In this case, we would
+need to catch the exception and update the state back to `Proposed`, but that case be more error
+prone than not updating the state in the first place. Having the state back to `Proposed` is
+important so that the wallet is left in a clean state and upon re-launched Daedalus the update can
+be downloaded again with no harm being done.
+
+## Determining the severity<a id="sec-5-2" name="sec-5-2"></a>
+
+For the proposed schema to succeed, we would need a function like the following:
+
+    updateSeverity :: UpdateProposal -> UpdateSeverity
+
+Such function would gather evidence from an `UpdateProposal` that this is a non-negotiable update.
+
+Note how we don't need to commit just yet whether or not we will treat hard-forks as `Mandatory`, as
+there are good arguments for people wanting to stick with an old version of the chain.
+
+## User stories<a id="sec-5-3" name="sec-5-3"></a>
+
+In order to check the validity of the proposed plan, let's try to run trough some of the common
+user stories for the update system.
+
+-   (1) As a Daedalus user, I want the wallet to prompt me if there is a new update available.
+
+At startup Daedalus would make a call to `GET /api/v1/update` to fetch the last available update. Note
+how this endpoint would **always** return a `WalletSoftwareUpdate`.
+
+In case this is the first time we are launching Daedalus and there are not updates available,
+we can issue a special "genesis" update record which `UpdateType` will be `Applied`,
+so that it will appear from the outside as no updates are available (which is what we want). In pseudo-code:
+
+``` haskell
+freshClientUpdate :: WalletSoftwareUpdate {
+freshClientUpdate = WalletSoftwareUpdate { 
+          wsuSoftwareVersion = ...
+        , wsuBlockVesion     = ...
+        , wsuScriptVersion   = ...
+        , wsuImplicit        = ...
+        , wsuVotesFor        = ...
+        , wsuVotesAgainst    = ...
+        , wsuPositiveStake   = ...
+        , wsuNegativeStake   = ...
+        , wsuSeverity        = Optional
+        , wsuUpdateState     = Applied
+        , wsuSha256Checksum  = ...
+        }
```

Comment:
I think originally, when I wrote the spec, this invariant was helping in the update process in case the user would abruptly hardquit whilst the update process is running, or when a new update arrived. I think, though, since I have separated the current update vs the next update, I think we might be able to go with your suggestion and return a `404`. Let me dwell on it, and I will upgrade the proposal accordingly or share my concerns here :)